astrobase/lcfit/transits.py

```diff
@@ -1658,3 +1677,639 @@ def nll(*args):
+
+    fitpklsavpath = os.path.join(
+        fitparamdir,
+        '{}_phased_{}_fit_empiricalerrs.pickle'.
+        format(identifier, fittype)
+    )
+
+    if os.path.exists(fitpklsavpath) and not overwriteexistingsamples:
+        maf_empc_errs = pickle.load(open(fitpklsavpath, 'rb'))
+
+    else:
+        maf_empc_errs = mandelagol_fit_magseries(
+                        sel_time, sel_flux, empirical_err,
+                        initfitparams, priorbounds, fixedparams,
+                        trueparams=cornerparams, magsarefluxes=True,
+                        sigclip=None, plotfit=phasedlcsavpath,
+                        plotcorner=cornersavpath,
+                        samplesavpath=samplesavpath, nworkers=nworkers,
+                        n_mcmc_steps=n_mcmc_steps, exp_time_minutes=30,
+                        eps=1e-6, n_walkers=500, skipsampling=False,
+                        overwriteexistingsamples=overwriteexistingsamples,
+                        mcmcprogressbar=mcmcprogressbar)
+
+        with open(fitpklsavpath, 'wb') as f:
+            pickle.dump(maf_empc_errs, f, pickle.HIGHEST_PROTOCOL)
+            LOGINFO('saved {:s}'.format(fitpklsavpath))
+
+    # fitfluxs, fittimes = _get_interp_fitfluxs(maf_empc_errs, sel_time)
+    # now plot the phased lightcurve
+    fitfluxs = maf_empc_errs['fitinfo']['fitmags']
+    fittimes = maf_empc_errs['magseries']['times']
+    fitperiod = maf_empc_errs['fitinfo']['finalparams']['period']
+    fitepoch = maf_empc_errs['fitinfo']['finalparams']['t0']
+
+    outfile = os.path.join(
+        fit_savdir,
+         "fancy_{}_phased_{}_fit_empiricalerrs.png".
+        format(identifier, fittype)
+    )
+
+    tdur_phase = t_dur_day/fitperiod
+
+    plot_phased_magseries(
+        sel_time, sel_flux, fitperiod, magsarefluxes=True, errs=None,
+        normto=False, epoch=fitepoch, outfile=outfile, sigclip=False,
+        phasebin=0.01, phasewrap=True, phasesort=True,
+        plotphaselim=[-n_transit_durations*tdur_phase,
+                      n_transit_durations*tdur_phase],
+        plotdpi=400, modelmags=fitfluxs, modeltimes=fittimes,
+        xaxlabel='Time from mid-transit [days]', yaxlabel='Relative flux',
+        xtimenotphase=True)
+
+    LOGINFO('made {}'.format(outfile))
+
+    #
+    # check if we have converged chains.
+    #
+    autocorrtime = maf_empc_errs['fitinfo']['autocorrtime']
+
+    if n_mcmc_steps < autocorrtime*50:
+        # autocorrtime estimate can't be trusted if above condition isn't true.
+        # past that condition, you're probably "converged" (though really each
+        # parameter takes different lengths of time to converge).  see
+        # https://dfm.io/posts/autocorr/, or the emcee docs.  might actually be
+        # converged if shorter, but to be safe, say it is not.
+        is_converged = False
+
+    else:
+        is_converged = True
+
+    return maf_empc_errs, is_converged
+
+
+def get_tess_limb_darkening_guesses(teff, logg):
```

Comment:
I think this would actually be better suited for services/limbdarkening.py or something similar. You can use the modtemplate.py file to make this file in the services subdir and then move over this function to that file. That way, if you want to add more bands, etc., you can just import the functions from there.

Reply:
done